### ISSUE TITLE
Why default configurations are being preferred over nested configuration?

### ISSUE THREAD

**Issue by aleksandergrzyb** [realm/SwiftLint#2665]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

It's more of a question than a bug report. I have a nested configuration file in my repository. When I run SwiftLint in the root of the repository, it first tries to find configuration file at root path, but there is none so it creates `Configuration` object with [default values](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Models/Configuration.swift#L73). Then it recursively searches for other configuration files. It finds one. Next it tries to merge those two – one with default values, second one with values from my nested configuration file. At [this line](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Extensions/Configuration+Merging.swift#L131) the code prefers `rules` in `self` (which are default rules) and not `configuration.rules` (which are rules defined in nested configuration file). This results with default rules configuration being chosen instead of ones from nested configuration file.

For example default configuration for `line_length` rule is `warning: 120` and `error: 200`. In my nested configuration file I defined configuration for `line_length` as `warning: 150` and `error: 150`. Current merging technique results with `warning: 120` and `error: 200` being chosen, not my nested configuration. Is that desired behavior?

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.30.1.

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Binary stored in repository.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

Yes.

```yml
line_length:
  warning: 150
  error: 150
```
* Which Xcode version are you using (check `xcode-select -p`)?

10.1.

**Comment by joeganley** [comment]
Similarly, if I have a repo with multiple source directories, and I put the .swiftlint.yml file at the top level of that repo and then run swiftlint in one of the source directories (where there is no .swiftlint.yml file), I would expect it to pick up the rules from the .swiftlint.yml in the repo's top level, but instead it seems to use all default rules.

**Comment by bsarrazin** [comment]
@aleksandergrzyb have you been able to find a solution/answer?
I am experiencing something similar (if not identical).
Reproducible project:
[Linter.zip](https://github.com/realm/SwiftLint/files/3244920/Linter.zip)

**Comment by aleksandergrzyb** [comment]
Yes, we did by introducing a "hack" to our `.swiftlint.yml` files. When you add the rule into `opt_in_rules` it gets respected. It's a flow in code, but it works. So for example if you want to enable `line_length` rule you need to define it like this:

```yml
opt_in_rules:
  - line_length

line_length: 150
```

Unfortunately it's repetitive, but this is the best what we've came up with.
